utils/dbt_format_utils.py

```diff
@@ -0,0 +1,365 @@
+
+    Args:
+        base_tables_path (str): The path to the directory containing the SQL files to be renamed.
+
+    Returns:
+        None. The function modifies the files in place by renaming them.
+    """
+    base_tables = [
+        f
+        for f in os.listdir(base_tables_path)
+        if os.path.isfile(os.path.join(base_tables_path, f))
+    ]
+    base_tables.sort()
+    for table_file in base_tables:
+        table_name = table_file.split(".")[0]
+        # Skip if not sql file
+        if ".sql" not in table_file:
+            print(f"{table_file} not a SQL file. Skipping.")
+            continue
+        # Skip if _base already in filename
+        if "_base.sql" in table_file:
+            print(f"- name: {table_name}")
+            continue
+
+        os.rename(
+            f"{base_tables_path}{table_file}",
+            f"{base_tables_path}{table_name}_base.sql",
+        )
+        print(f"- name: {table_name}_base")
+
+
+def create_matview_dbt_files_from_base(base_tables_path, output_path):
+    """
+    Creates DBT materialized view files based on base tables by removing Airbyte specific columns
+    and writing SQL to a new file. (SQL will be a select all columns except Airbyte columns)
+
+    Args:
+        base_tables_path (str): The path to the directory containing the base tables.
+        output_path (str): The path to the directory where the new files will be written.
+
+    Returns:
+        None. The function writes the new files to the specified output directory.
+
+    """
+
+    # Make sure path structure is consistent
+    base_tables_path = (
+        base_tables_path if base_tables_path[-1] != "/" else base_tables_path[:-1]
+    )
+    output_path = output_path if output_path[-1] != "/" else output_path[:-1]
+    # Check if output exists
+    isExist = os.path.exists(output_path)
+    if not isExist:
+        # Create a new directory because it does not exist
+        os.mkdir(output_path)
+        print(f"{output_path} is created!")
+
+    base_tables = [
+        f
+        for f in os.listdir(base_tables_path)
+        if os.path.isfile(os.path.join(base_tables_path, f))
+    ]
+    for table in base_tables:
+        with open(f"{base_tables_path}/{table}", "r") as infile:
+            with open(f"{output_path}/{table.replace('_base','')}", "w") as outfile:
+                write_to_file = False
+                table_name = table.split(".")[0]
+
+                # Only write lines from base tables that include SQL Statement ( SELECT -> FROM )
+                # Excludes _airbyte columns
+                for line in infile:
+                    if "select" in line:
+                        write_to_file = True
+                    if "from {{" in line and "--" not in line:
+                        write_to_file = False
+                        outfile.write(f"from {{{{ source('cta','{table_name}') }}}}")
+                    if write_to_file:
+                        if not "_airbyte" in line:
```

Comment:
@jitoquinto I think we want to keep the `_airbyte_*_hashid` fields in there since those are supposed to be the unique primary keys, at least by default until we go in and figure out the actual UPKs for each table - can we modify this logic so it yeets all the _airbyte fields _except_ the hashid?

Reply:
Actually I wonder if the easiest strat is to just do something like

```
select
    *
EXCEPT _airbyte_ab_id, _airbyte_emitted_at, _airbyte_normalized_at
from .....
```

🤔 whatever you think is easiest here is fine with me, but we do want them hashids 👀

Reply:
@emily-flambe we want this hash_id even for matviews?